src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -69,8 +69,29 @@ getLedgerInfo(AbstractLedgerTxn& ltx, Application& app,
     info.min_temp_entry_ttl =
         sorobanConfig.stateArchivalSettings().minTemporaryTTL;
     info.max_entry_ttl = sorobanConfig.stateArchivalSettings().maxEntryTTL;
-    info.cpu_cost_params = toCxxBuf(sorobanConfig.cpuCostParams());
-    info.mem_cost_params = toCxxBuf(sorobanConfig.memCostParams());
+
+    auto cpu = sorobanConfig.cpuCostParams();
+    auto mem = sorobanConfig.memCostParams();
+
+// The host expects this vector to be the size of the # of cost types in the
```

Comment:
This seems kind of hacky, maybe we should have protocol-based checks on the host side? I'm not sure we should be using new cost types in protocol 20.

Reply:
Yeah I agree this is pretty hacky, but [cost_models](https://github.com/stellar/rs-soroban-env/blob/f8d3bafa19b934741efbb6f8ef3311696b3556de/soroban-env-host/src/budget/dimension.rs#L18C16-L18C27) is an array, so the length is set at compile time. We could dynamically allocate it, but I went with this path instead. I'm not opposed to fixing this in env though.

Reply:
Yeah, I think it should probably be a vector. But I think we can try to refactor that as a followup after p21 is out.

Reply:
https://github.com/stellar/rs-soroban-env/issues/1388